app/screens/settings/display_theme/display_theme.tsx

```diff
@@ -26,48 +27,54 @@ type DisplayThemeProps = {
 const DisplayTheme = ({allowedThemeKeys, componentId, currentTeamId, currentUserId}: DisplayThemeProps) => {
     const serverUrl = useServerUrl();
     const theme = useTheme();
-    const initialTheme = useRef(theme);
-    const [newTheme, setNewTheme] = useState<string | undefined>(undefined);
+    const [currentTheme, setCurrentTheme] = useState(theme);
+    const [customTheme, setCustomTheme] = useState(theme.type?.toLowerCase() === 'custom' ? theme : undefined);
 
     const close = () => popTopScreen(componentId);
 
-    const setThemePreference = useCallback(() => {
-        const allowedTheme = allowedThemeKeys.find((tk) => tk === newTheme);
-        const themeJson = Preferences.THEMES[allowedTheme as ThemeKey] || initialTheme.current;
+    const setThemePreference = useCallback(async (theTheme: string) => {
+        const allowedTheme = allowedThemeKeys.find((tk) => tk === theTheme);
+        const themeJson = Preferences.THEMES[allowedTheme as ThemeKey] || customTheme;
 
         const pref: PreferenceType = {
             category: Preferences.CATEGORIES.THEME,
             name: currentTeamId,
             user_id: currentUserId,
             value: JSON.stringify(themeJson),
         };
-        savePreference(serverUrl, [pref]);
-    }, [allowedThemeKeys, currentTeamId, currentUserId, serverUrl, newTheme]);
+        await savePreference(serverUrl, [pref]);
+
+        setCurrentTheme(themeJson);
+
+    }, [allowedThemeKeys, currentTeamId, currentUserId, serverUrl, customTheme]);
+
+    const handleSelectTheme = usePreventDoubleTap(useCallback((themeSelected: string) => {
```

Comment:
I guess even if we prevent the double tap, this may generate other issues on slow connections.

Shouldn't we just use a loading pattern? i.e.
```
const foo = useCallback(async () => {
    if (loading) {
        return;
    }
    setLoading(true);
    await doSomething();
    setLoading(false);
}, [loading]);
```

(Maybe it make sense to pair it with the prevent double tap)

Reply:
@larkox in reality, we should also have a try..catch, in order to catch any network issue that comes out of this as well.

But again, the original problem is changing theme repeatedly.  We can spend a lot more time making sure this screen is solid as can be.

Reply:
You're right, if it fails, then the theme doesn't get switched.  So try..catch is not necessary.

And yes, if the user taps the second theme outside of the 750ms default, then yes, it could be causing the same race condition as well.  But introducing loading indicator would involved the design team and as mentioned, is out of the scope of this ticket.  I'm happy to create a new ticket for it.

Reply:
@lieut-data